bittensor/synapse.py

```diff
@@ -25,15 +25,25 @@ def synapse_key(self) -> str:
     def set_synapse_key(self, key):
         self._synapse_key = key
 
-    def deepcopy(self):
+    def deepcopy(self, config = None):
         """ Returns a copy of this synapse by passing the model params to load_state_dict.
 
+            config: (:obj:`config.class`, `optional`, defaults to model.config): 
+                    model config used to re-init the model.
+
             Returns:
                 synapse_copy (:obj:`self.__class__`, `required`): 
                     Deep copy synapse object.
         """
         SynapseClass = self.__class__
-        synapse_copy = SynapseClass()
+        
+        model_config = None
+        if config == None:
+            # If no passed config, try model config.
+            if self.config == None:
+                raise ValueError('Deep copy requires a passed model config object or a member model.config')
+            model_config = self.config
+        synapse_copy = SynapseClass(model_config)
```

Comment:
this looks weird. `config` vs `self.config` can be confusing. Maybe rename config to `synapse_config`?